lib/avro_turf/connection_wrapper_with_token_auth.rb

```diff
@@ -39,7 +43,9 @@ def initialize(url,
 
   def with_connection
     refresh_token if token_needs_refresh?
-    result = super
+    result = connection_pool.with do |conn|
```

Comment:
I think these has to go into the super class `AvroTurf::ConnectionWrapper`, where we actually create, store and yield the connection.

At the same time, we might want to also introduce the same options we have for the Espresso client: persistent connections and retry options.

Reply:
makes sense, persistent connections is already there (but hardcoded). Agree on retries 👍

Reply:
lol I had already added retries sorry! it's [here](https://github.com/meetcleo/avro_turf/blob/master/lib/avro_turf/confluent_schema_registry.rb) because I added them only for gets

Reply:
and done!